driver-core/src/test/functional/com/mongodb/client/model/expressions/ExpressionsFunctionalTest.java

```diff
@@ -0,0 +1,646 @@
+        StringExpression thenC = Expressions.ofString("C");
+        StringExpression elseD = Expressions.ofString("default");
+
+        // Expressed, inconveniently, as a cond-sequence:
+        assertExpression(
+                "C",
+                fal.cond(thenA,
+                        fal.cond(thenB,
+                                tru.cond(thenC, elseD)))
+        );
+
+        // A switch is similar to an Optional chain.
+        assertExpression(
+                "C",
+                ofNull()
+                        .ifNull(tru.eq(fal).cond(thenA, ofNull()))
+                        .ifNull(tru.eq(fal).cond(thenB, ofNull()))
+                        .ifNull(tru.eq(tru).cond(thenC, ofNull()))
+                        .ifNull(elseD)
+        );
+        // This would not work, since individual branches can produce null, and
+        // there is no other suitable "missing" or "empty" value.
+
+        // The above suggests something like the following, which requires a
+        // separate, non-Expression class, which is produced via the lambda:
+        BooleanExpression value = Expressions.ofBoolean(true);
+        value.switchMap((e, init) -> init
+                .caseEq(fal, thenA)
+                .caseEq(fal, thenB)
+                .caseEq(tru, thenC)
+                .defaults(elseD)
+        );
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/meta/
+    }
+
+
+    @Test
+    public void comparisonExpressionsTest() {
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/#comparison-expression-operators
+        // (Complete as of 6.0)
+        // Comparison expressions are on the generic Expression class.
+
+        BooleanExpression tru = Expressions.ofBoolean(true);
+        StringExpression str = Expressions.ofString("string");
+        Expression nul = Expressions.ofNull();
+
+        // equality:
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/eq/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/ne/
+
+        // comparison:
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/gt/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/gte/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/lt/
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/lte/
+
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/cmp/ (89)
+        // TODO implement cmp as a three-branch cond?
+
+        // https://www.mongodb.com/docs/manual/reference/operator/aggregation/ifNull/
+        // This is similar to orElse, but treats nulls as empty Optionals
+        // TODO the last position is auto-filled with null
+        // TODO this diverges from the MQL
+        // TODO just as with 1.add(2).add(3), multi-param ifNull should be chained: a.ifNull(b).ifNull(c)
```

Comment:
We should fill the last "mandatory" position with null. This should also be the behaviour of the server ifNull.

Reply:
For this, we use the isIntegerOr (etc.) methods. There is no ifNull (though we use it internally).